File: 0905-sort-array-by-parity/0905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        //O(n) O(n)
        int n=nums.length;
        int []ans=new int[n];
        int ev=0,od=ans.length-1;
        for(int i=0;i<n;i++){
            if(nums[i]%2==0){
                ans[ev]=nums[i];
                ev++;
            }
            else{
                ans[od]=nums[i];
                od--;
            }
             
        }
        return ans;
    }
}

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nn) {
        int i=0,j=0,n=nn.size();
        for(i=0;i<n;i++){
            if(nn[i]%2==0){
            swap(nn[i],nn[j]);
            j++;
            }
        }
        return nn;
    }
};